contracts/secret-storage/SecretManager.sol

```diff
@@ -498,20 +514,28 @@ contract SecretManager is
         if(!userStorage[_secretId].selectedEnclaves[enclaveIndex].hasAcknowledgedStore)
             return;
 
-        _updateUsdcDepositPostDeadMark(_secretId, _enclaveAddress, enclaveIndex);
+        uint256 deadStorePayment = _getStoreUsdcPayment(_secretId, _enclaveAddress, enclaveIndex);
+        userStorage[_secretId].usdcDeposit -= deadStorePayment;
 
-        _replaceStore(_secretId, _enclaveAddress, enclaveIndex, true);
+        uint256 sizeLimit = userStorage[_secretId].sizeLimit;
+        if(block.timestamp > userStorage[_secretId].endTimestamp) {
+            _removeSelectedEnclave(_secretId, enclaveIndex);
+            if(userStorage[_secretId].selectedEnclaves.length == 0)
+                _refundExcessDepositAndRemoveStore(_secretId);
+        }
+        else
+            _replaceStore(_secretId, _enclaveAddress, enclaveIndex, true);
 
         // TODO: slash prev enclave(who's recipient)
-        SECRET_STORE.slashEnclave(_enclaveAddress, userStorage[_secretId].sizeLimit, STAKING_PAYMENT_POOL);
+        SECRET_STORE.slashEnclave(_enclaveAddress, sizeLimit, STAKING_PAYMENT_POOL);
     }
 
     /// @dev It removes the dead store payment from the usdc deposit of the secret.
-    function _updateUsdcDepositPostDeadMark(
+    function _getStoreUsdcPayment(
```

Comment:
what is this function is supposed to return?

According to me it should be payment for the (endTime -  lastAliveTime), no?

Reply:
I have modified this function.
Now, It updates usdc deposit based on the payment to the selected store for a specific secret.
It is executed when - 
1. alive check is submitted for terminated secretId
2. dead check
3. remove secret is called post termination